python/cuspatial/cuspatial/geometry/geoarrowbuffers.py

```diff
@@ -0,0 +1,558 @@
+        result = MultiPointArray(base.xy, base.offsets, base.z)
+        return result
+
+
+class PolygonArray(OffsetArray):
+    def __init__(self, xy, polys, rings, mpolys, z=None, data_locale=cudf):
+        """
+        The GeoArrow column format for Polygons uses the same scheme as the
+        format for LineStrings - MultiPolygons and Polygons from the same
+        GeoSeries (or another data source) are stored in the same contiguous
+        buffer. Rings are stored in the offsets array, exterior/interior
+        polygons are stored in the `polys` array in shapefile format, and the
+        `mpolys` array of pairs determines which polys are members of
+        MultiPolygons.
+        """
+        # PolygonArray uses the offsets buffer for rings!
+        super().__init__(xy, rings, z, data_locale=data_locale)
+        self.polys = polys
+        self.mpolys = mpolys
+
+    @property
+    def rings(self):
+        """
+        Each polygon in the GeoSeries is specified in the rings buffer, which
+        follows the same format as `offsets` in OffsetArray.
+        """
+        return self.offsets
+
+    @rings.setter
+    def rings(self, rings):
+        self.offsets = self._serialize(rings)
+
+    @property
+    def polys(self):
+        """
+        Polygons are specified according to the shapefile format. The first
+        ring in any polygon is its exterior ring, all subsequent rings are
+        interior rings. The `polys` column specifies which rings comprise each
+        polygon in the GeoSeries.
+        """
+        return self._polys
+
+    @polys.setter
+    def polys(self, polys):
+        self._polys = self._serialize(polys)
+
+    @property
+    def mpolys(self):
+        """
+        Polygons that fall within bounds constrained by each pair of values in
+        this column are members of a MultiPolygon. Each pair of values
+        specifies a single MultiPolygon. All MultiPolygons in the GeoSeries
+        are contained here.
+        """
+        return self._mpolys
+
+    @mpolys.setter
+    def mpolys(self, mpolys):
+        self._mpolys = self._serialize(mpolys)
+
+    def __repr__(self):
+        result = (
+            f"{super().__repr__()}"
+            f"polys:\n"
+            f"{self.polys}\n"
+            f"rings:\n"
+            f"{self.rings}\n"
+            f"mpolys:\n"
+            f"{self.mpolys}\n"
+        )
+        return result
+
+    def copy(self, deep=True):
+        """
+        See CoordinateArray
+        """
+        result = super().copy(deep)
+        result.polys = self.polys.copy(deep)
+        result.rings = self.rings.copy(deep)
+        result.mpolys = self.mpolys.copy(deep)
+        return result
+
+    def __len__(self):
+        mlength = np.array(
+            [
+                self._mpolys[2 * i - 1] - self._mpolys[2 * i - 2] - 1
+                for i in range(1, len(self._mpolys) // 2 + 1)
+            ],
+            dtype="int64",
+        ).sum()
```

Comment:
We're doing iter->list->array->sum here; is it simpler/faster to just do iter->sum?

```
sum(
    self._mpolys[2 * i - 1] - self._mpolys[2 * i - 2] - 1
    for i in range(1, len(self._mpolys) // 2 + 1)
)
```

Reply:
Sounds good!